test/Classes/IntersectionTypeConstructorDependency.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\Di\Classes;
+
+use Countable;
+use Iterator;
+
+class IntersectionTypeConstructorDependency
+{
+    public function __construct(private Iterator&Countable $iter)
```

Comment:
I wonder if we can pin to a newer phpcs dependency to fix this?

Reply:
Is this something I can test within laminas-di?

Reply:
Yeah, phpcs should be in require-dev, no?

Reply:
I think not explicitly, as `"laminas/laminas-coding-standard": "~2.3.0"` is used if I'm correct on this?

Reply:
Yeah, but we can add phpcs explicitly, to force a newer version

Reply:
Forgive me for my lack of knowledge, but would this addition to the `composer.json` and `composer.lock` suffice?

Reply:
Could it be that the intersection types are not yet supported, there seems to be an issue [#3479](https://github.com/squizlabs/PHP_CodeSniffer/issues/3479) for intersection type support, related PR [3581](https://github.com/squizlabs/PHP_CodeSniffer/pull/3581)?

Reply:
I guess we could perhaps exclude the file in phpcs.xml.dist then?